java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspTemplateUI.java

```diff
@@ -215,12 +238,14 @@ public CompletionStage<Pair<DataFolder,String>> apply(String path) {
                     }
                     targetPath.getParentFile().mkdirs();
                     FileObject fo = FileUtil.toFileObject(targetPath.getParentFile());
-                    if (fo == null || !fo.isFolder()) {
+                    if (fo == null || !fo.isFolder() || !targetPath.getParentFile().canWrite() || !SourceVersion.isName(targetPath.getName())) {
```

Comment:
I am not quite sure I read the patch correctly, but this is the folder where the project will be created. And the name of this folder does not necessarily need to follow the Java name rules(?). I mean, I can have a Java project in a directory called `my-current-java-project`, and there's no problem in that? Or do I miss something here?

Reply:
Right it will be the name of the folder where the project will be created . We need the java name rules because that is also used for the name of the main class . So if the path is `/Users/Projects/my-current-java-project` it will try to create a project with the main class name  as `my-current-java-project`.

Reply:
this still checks java constraints for non-java projects.